lib/plugins/blocks.js

```diff
@@ -457,20 +456,25 @@ function inject (bot, { version, storageBuilder }) {
     bot.world = new World(null, storageBuilder ? storageBuilder({ version: bot.version, worldName: dimensionToFolderName(packet.dimension) }) : null).sync
     startListenerProxy()
   })
-
+  let listenerState
   function startListenerProxy () {
+    if (listenerState !== undefined) {
+      bot.removeAllListeners('removeListener')
+    }
     // standardized forwarding
     const forwardedEvents = ['blockUpdate', 'chunkColumnLoad', 'chunkColumnUnload']
     for (const event of forwardedEvents) {
       bot.world.on(event, (...args) => bot.emit(event, ...args))
     }
     // custom forwarder for custom events
     const blockUpdateRegex = /blockUpdate:\(-?\d+, -?\d+, -?\d+\)/
-    bot.on('newListener', (event, listener) => {
-      if (blockUpdateRegex.test(event)) {
-        bot.world.on(event, listener)
-      }
-    })
+    if (listenerState === undefined) {
+      listenerState = bot.on('newListener', (event, listener) => {
```

Comment:
you need to store the listener (the function) if you want to be able to remove this one specifically (and avoid removing all listeners)
```js
listenerState = (event, listener) => { /* the function */}
bot.on('newListener', listenerState)
```
then you can do `bot.removeListener('newListener', listenerState)`

And do the same for the removeListener event

Reply:
I was experimenting with this, however it wasn't working properly with `bot.removeListener`

Reply:
`js
  function startListenerProxy () {
    let listener = (event, listener) => {
      if (blockUpdateRegex.test(event)) {
        bot.world.on(event, listener)
      }
    }
    bot.removeListener('removeListener', listener)
    bot.removeListener('newListener', listener)
    // standardized forwarding
    const forwardedEvents = ['blockUpdate', 'chunkColumnLoad', 'chunkColumnUnload']
    for (const event of forwardedEvents) {
      bot.world.on(event, (...args) => bot.emit(event, ...args))
    }
    // custom forwarder for custom events
    const blockUpdateRegex = /blockUpdate:\(-?\d+, -?\d+, -?\d+\)/
    bot.on('newListener', listener)
    bot.on('removeListener', listener)
  }
`

Reply:
This is the code that I was trying to use, however it doesn't seem to remove them properly

Reply:
try with
```js
  let listenerState
  function startListenerProxy () {
    if (listenerState) {
      // custom forwarder for custom events
      bot.removeListener('removeListener', listenerState)
      bot.removeListener('newListener', listenerState)
    }
    // standardized forwarding
    const forwardedEvents = ['blockUpdate', 'chunkColumnLoad', 'chunkColumnUnload']
    for (const event of forwardedEvents) {
      bot.world.on(event, (...args) => bot.emit(event, ...args))
    }
    const blockUpdateRegex = /blockUpdate:\(-?\d+, -?\d+, -?\d+\)/
    listenerState = (event, listener) => {
      if (blockUpdateRegex.test(event)) {
        bot.world.on(event, listener)
      }
    }
    bot.on('newListener', listenerState)
    bot.on('removeListener', listenerState)
  }
```

Reply:
Yep that code works I believe.